src/class/pmix_object.h

```diff
@@ -338,11 +336,11 @@ static inline pmix_object_t *pmix_obj_new_debug_tma(pmix_class_t *type, pmix_tma
         do {                                                                         \
             assert(NULL != ((pmix_object_t *) (object))->obj_class);                 \
             assert(PMIX_OBJ_MAGIC_ID == ((pmix_object_t *) (object))->obj_magic_id); \
-            pmix_obj_update((pmix_object_t *) (object), 1);                          \
+            ((pmix_object_t *) (object))->obj_reference_count++;                     \
```

Comment:
We'd probably want to do a bit of performance analysis, but the other option, which is where I was leaning, is to just have a pthread_lock in the base object and protect the ref counting that way.  That way, if you do have thread contention somewhere accidentally, you don't have silent data corruption.

Reply:
Done - we aren't a performant library, so the cost of the lock is somewhat irrelevant.